creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -1194,35 +1195,44 @@ class SnapbackSM {
    * @notice this will issue sync to healthy secondary and update replica set away from unhealthy secondary
    */
   async aggregateReconfigAndPotentialSyncOps(nodeUsers, unhealthyPeers) {
+    // Parallelize calling this._aggregateOpsWithQueriedSpIds on chunks of 500 nodeUsers at a time
+    const nodeUserBatches = _.chunk(
+      nodeUsers,
+      AGGREGATE_RECONFIG_AND_POTENTIAL_SYNC_OPS_BATCH_SIZE
+    )
+    const resultBatches = []
+    while (nodeUserBatches.length) {
+      const nodeUserBatch = nodeUserBatches.shift()
+      const resultBatch = await Promise.allSettled(
+        nodeUserBatch.map((nodeUser) =>
+          this._aggregateOpsWithQueriedSpIds(nodeUser, unhealthyPeers)
+        )
+      )
+      resultBatches.push(resultBatch)
+    }
+
+    // Combine each batch's requiredUpdateReplicaSetOps and potentialSyncRequests
+    const results = _.flatten(resultBatches)
```

Comment:
super nit - i think it may be faster to just do `resultBatches.push(...resultBatch)` and remove this flatten call

perf really not a problem here but that may or may not be more readable - your call tho

Reply:
I like that idea 👍

Reply:
I actually think using lodash is cleaner since we dont have to loop thru the batches, spread the input results, and append it to a greater array

Reply:
I think Sid was just saying on line 1211 I can push the elements of the array (spread it) instead of pushing the array. then there's no need to flatten after. sounds good to me, wdyt?

Reply:
although I think it's better to do it as concat since everywhere else in the code uses concat and concat is slightly faster for this performance-critical portion

Reply:
oh, in that case yes, lets go with sid's suggestion!